connectivity/tests/common.go

```diff
@@ -48,3 +54,81 @@ func hasAllLabels(labelsContainer labelsContainer, filters map[string]string) bo
 	}
 	return true
 }
+
+type retryCondition struct {
+	podLabels map[string]string
+	all       bool
+	destPort  uint32
+	destIP    string
+}
+
+// CurlOptions returns curl retry option or empty slice depending on retry conditions
+// if all value set it means always return retry options
+// for pod-to-cidr all or destIP is checked
+// for pod-to-world all, destPort and destPort + pod label is checked
+func (rc *retryCondition) CurlOptions(peer check.TestPeer, ipFam check.IPFamily, pod check.Pod, params check.Parameters) []string {
+	opts := []string{}
+
+	if connectRetry := params.Retry; connectRetry > 0 {
+		opts = append(opts, "--retry", strconv.FormatInt(int64(connectRetry), 10),
+			"--retry-all-errors") // add --retry-all-errors to retry on all possible errors
+
+		// only add --retry-delay if --retry is enabled
+		if retryDelay := params.RetryDelay.Seconds(); retryDelay > 0.0 {
+			opts = append(opts, "--retry-delay", strconv.FormatFloat(retryDelay, 'f', -1, 64))
+		}
+	}
+
+	// no retry options given return early
+	if len(opts) == 0 {
+		return opts
+	}
+	// all conditions is set return early
+	if rc.all {
+		return opts
+	}
+	// destIP is checked with pod-to-cidr cases
+	if peer.Address(ipFam) == rc.destIP {
+		return opts
+	}
+	// destPort and podLabel are checked with pod-to-world case
+	if peer.Port() == rc.destPort {
+		for n, v := range rc.podLabels {
+			if !pod.HasLabel(n, v) {
+				return []string{}
+			}
+		}
+		return opts
+	}
+
+	return []string{}
+}
+
+type RetryOption func(*retryCondition)
+
+func WithRetryAll() RetryOption {
+	return func(ef *retryCondition) {
+		ef.all = true
+	}
+}
+
+func WithRetryDestIP(ip string) RetryOption {
+	return func(ef *retryCondition) {
+		ef.destIP = ip
+	}
+}
+
+func WithRetryDestPort(port uint32) RetryOption {
+	return func(ef *retryCondition) {
+		ef.destPort = port
+	}
+}
+
+func WithRetryPodLabel(name, val string) RetryOption {
+	return func(ef *retryCondition) {
+		if ef.podLabels == nil {
+			ef.podLabels = map[string]string{}
+		}
+		ef.podLabels[name] = val
+	}
+}
```

Comment:
Nit: change the receiver name from `ef` to `rc`:

```suggestion
func WithRetryAll() RetryOption {
	return func(rc *retryCondition) {
		rc.all = true
	}
}

func WithRetryDestIP(ip string) RetryOption {
	return func(rc *retryCondition) {
		rc.destIP = ip
	}
}

func WithRetryDestPort(port uint32) RetryOption {
	return func(rc *retryCondition) {
		rc.destPort = port
	}
}

func WithRetryPodLabel(name, val string) RetryOption {
	return func(rc *retryCondition) {
		if rc.podLabels == nil {
			rc.podLabels = map[string]string{}
		}
		rc.podLabels[name] = val
	}
}
```

Reply:
First I used and other name for retryCondition which it short form was ef 😄 
good catch.